### DISCUSSION TITLE
error in roslaunch carla0.9.10 and ubuntu18.04

### DISCUSSION THREAD

**Discussion by cepheus948** [carla-simulator/carla#6015] [Q&A]
when I run roslaunch carla_ros_bridge carla_ros_bridge.launch, there is an error called 'CARLA python module version 0.9.13 required. Found: 0.9.10'. I don't know why it requires Carla0.9.13. Actually, I use carla 0.9.10 and corresponding version of carla-ros-bridge.I use python 2.7 and ROS1 melodic. Is there anything wrong with the versions and compatibility? Thanks for the answers!

[FATAL] [1670312604.160708, 0.000000]: CARLA python module version 0.9.13 required. Found: 0.9.10
[INFO] [1670312604.166333, 0.000000]: Shutting down...
Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rospy/core.py", line 572, in signal_shutdown
    h()
  File "/home/cee/carla-ros-bridge/catkin_ws/src/ros-bridge/carla_ros_bridge/src/carla_ros_bridge/bridge.py", line 349, in destroy
    self.shutdown.set()
AttributeError: 'CarlaRosBridge' object has no attribute 'shutdown'
================================================================================REQUIRED process [carla_ros_bridge-2] has died!
process has died [pid 15739, exit code 1, cmd /home/cee/carla-ros-bridge/catkin_ws/src/ros-bridge/carla_ros_bridge/src/carla_ros_bridge/bridge.py __name:=carla_ros_bridge __log:=/home/cee/.ros/log/a0be54a4-7539-11ed-8b03-584120ee5b54/carla_ros_bridge-2.log].
log file: /home/cee/.ros/log/a0be54a4-7539-11ed-8b03-584120ee5b54/carla_ros_bridge-2*.log
Initiating shutdown!
================================================================================
[carla_ros_bridge-2] killing on exit
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done

**Comment by zhangjw151** [comment]
我遇到了和你一样的问题，我在roslaunch yahboomcar_bringup bringup.launch 时，出现了Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rospy/core.py", line 572, in signal_shutdown
    h()
  File "/home/zhang/driver_ws/src/yahboomcar_bringup/scripts/Mcnamu_driver.py", line 45, in cancel
    self.imuPublisher.unregister()
AttributeError: yahboomcar_driver instance has no attribute 'imuPublisher'
================================================================================REQUIRED process [driver_node-4] has died!
process has finished cleanly
log file: /home/zhang/.ros/log/401dde6c-2547-11ee-b64f-000c29fa294c/driver_node-4*.log
Initiating shutdown!
================================================================================

**Comment by Anhar2** [comment]
Have you found the solution?

**Comment by yuan-mingxi** [comment]
same problem have anyone save it？

**Reply by Anhar2** [reply]
I'm trying to track the method "shutdown" and I cannot find it at any script, could you find any solution yet?

**Reply by yuan-mingxi** [reply]
Sorry for can not help you .
after i search for help from Internet and my teacher,but no useful soultion. finally decide to use PYTHON to test carla project
instead of ros-bridge.   My teacher told me the problem Maybe the pyhon enviroment didn't match.i use python 3.8 and ubuntu 20.04. if you find solution pls tell me. Thanks!

**Comment by yuan-mingxi** [comment]
邮件已收到，谢谢！（来自QQ邮箱自动回复邮件）

**Comment by pravinkmr26** [comment]
look for the file CARLA_VERSION under ros-bridge directory on your workspace, change the version in that file to the desired version, say 0.9.10 in your case. see if that helps
